### PR TITLE
config reader for auth configuration

### PR DESCRIPTION
**Pull request by MXPOL**

### CODE REVIEW

File: packages/external-db-config/lib/readers/common_config_reader.js [hunk 1]
```diff
@@ -4,8 +4,8 @@ class CommonConfigReader {
     constructor() { }
 
     readConfig() {
-        const { CLOUD_VENDOR, TYPE, REGION, SECRET_NAME } = process.env
-        return { vendor: CLOUD_VENDOR, type: TYPE, region: REGION, secretId: SECRET_NAME }
+        const { CLOUD_VENDOR, AUTH_VENDOR, TYPE, REGION, SECRET_NAME } = process.env
+        return { vendor: CLOUD_VENDOR, authVendor: AUTH_VENDOR, type: TYPE, region: REGION, secretId: SECRET_NAME }
```
Comment (MXPOL) [inline packages/external-db-config/lib/readers/common_config_reader.js:8]: @noam-almog  not sure if we need to verify `AUTH_VENDOR` variable (check that it was supplied) 
Will there always be authorization?
WDYT?

Comment (noam-almog) [reply packages/external-db-config/lib/readers/common_config_reader.js:8]: auth vendor is the same as vendor, no ?

Comment (MXPOL) [reply packages/external-db-config/lib/readers/common_config_reader.js:8]: We agreed that these would be 2 different things …

File: packages/external-db-config/lib/readers/aws_auth_config_reader.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+const { SecretsManagerClient, GetSecretValueCommand } = require('@aws-sdk/client-secrets-manager')
+const { checkRequiredKeys } = require('../utils/config_utils')
+
+const EmptyAWSAuthConfig = { callbackurl: '', clientid: '', clientsecret: '', clientdomain: '' }
+
+class AwsAuthConfigReader {
```
Comment (noam-almog) [inline packages/external-db-config/lib/readers/aws_auth_config_reader.js:6]: do you want to define a different secret key for auth ? 
i want to use the same key as before but auth will be a sub-property in the current config.

File: packages/external-db-config/lib/readers/common_config_reader.js (hunk 1 above)
Comment (noam-almog) [reply packages/external-db-config/lib/readers/common_config_reader.js:8]: why ? it's the same value ? aws/gcp/azure, no ?

File: packages/external-db-config/lib/readers/aws_auth_config_reader.js (hunk 2 above)
Comment (MXPOL) [reply packages/external-db-config/lib/readers/aws_auth_config_reader.js:6]: I can't set the value for `clientsecret`, it's a secret that generated when I create the login screen in AWS/GCP

Comment (MXPOL) [reply packages/external-db-config/lib/readers/aws_auth_config_reader.js:6]: image

**Pull request merged**